django_walletpass/services.py

```diff
@@ -4,11 +4,16 @@
 
 from aioapns import APNs, NotificationRequest
 from aioapns.exceptions import ConnectionClosed
+
 from django_walletpass.models import Registration
 from django_walletpass.settings import dwpconfig as WALLETPASS_CONF
+from django_walletpass.signals import TOKEN_UNREGISTERED
 
 logger = logging.getLogger('walletpass.services')
 
+async def send_notification_result_signal(notification_request, notification_result):
+    await TOKEN_UNREGISTERED.send(notification_request, notification_result)
```

Comment:
Maybe you are looking for this new django 5.0 feature: https://docs.djangoproject.com/en/5.1/topics/signals/#sending-signals which introduces a new asend() func. Keep in mind that calling await on a non async func will be result in a runtime err

Also, keep in mind that this new feature is only allowed in django > 5.0 so maybe an alternative for django < 5.0 should be provided in this new feature

Reply:
thx for pointing me towards that. I was already aware of the DJ5 change, but saw some SO doing similar. Looks like I have badly copied that code and the tests didn't cover this error :/

I was considering using the `sync_to_async` decorator for now and add a `# FIXME` comment to switch to a plain async function when we specify `Django>5` at some point in the future. Seems to me at least better than having two functions. or should I add a `asend_notification_result_signal` in parallel and a version switch? what would you prefer?

Reply:
I feel comfortable with the sync_to_async approach 👍

Reply:
@NachE done: https://github.com/Develatio/django-walletpass/pull/42/commits/266daf8b41ba67469364fa27fdab163aa29e14b2